pyanaconda/payload/dnf/payload.py

```diff
@@ -1928,9 +1938,19 @@ def _add_treeinfo_repositories(self, install_tree_url, base_repo_url=None):
                     repo = RepoData(name=repo_md.name, baseurl=repo_md.path,
                                     install=False, enabled=repo_enabled)
                     repo.treeinfo_origin = True
+                    log.debug("Adding new treeinfo repository %s", repo_md.name)
                     self.add_repo(repo)
 
-        return install_tree_url
+    def _cleanup_old_treeinfo_repositories(self):
```

Comment:
I think we should do this after reset. The repositories won't be deleted if the source is changed to for example to CDN.

Reply:
I'm not sure I understand you. When we do this after reset then we have to do the reset again because these repositories will be added back.

Reply:
Fixed.